assets/app/scripts/controllers/attachPersistentVolumeClaim.js

```diff
@@ -0,0 +1,280 @@
+          return true;
+        };
+
+        var volumeNameIsUnique = function(name, podTemplate) {
+          if (podTemplate.spec.volumes) {
+            for (var i = 0; i < podTemplate.spec.volumes.length; i++) {
+              var volume = podTemplate.spec.volumes[i];
+              if (volume.name === name) {
+                return false;
+              }
+            }
+          }
+          return true;
+        };
+
+        var volumeMountPathIsUnique = function(name, mountPath, podTemplate) {
+          if (podTemplate.spec.containers) {
+            for (var i = 0; i < podTemplate.spec.containers.length; i++) {
+              var container = podTemplate.spec.containers[i];
+              if ($scope.attach.containers.all || $scope.attach.containers.individual[container.name]) {
+                if (container.volumeMounts) {
+                  for (var j = 0; j < container.volumeMounts.length; j++) {
+                    var volumeMount = container.volumeMounts[j];
+                    if (volumeMount.mountPath === mountPath && name !== volumeMount.Name) {
+                      return false;
+                    }
+                  }
+                }
+              }
+            }          
+          }
+          return true;
+        };
+
+        // breadcrumb must react depending on what we are attaching to (deployment or dc)
+        var rebuildBreadcrumb = function() {
+          $scope.breadcrumbs.splice(1, 0, {
+            title: "Deployments",
+            link: "project/" + $routeParams.project + "/browse/deployments"
+          });
+
+          var deploymentConfig = $scope.attach.deploymentConfig;
+          if (deploymentConfig) {
+            $scope.breadcrumbs.splice(2, 0, {
+              title: deploymentConfig.metadata.name,
+              link: "project/" + $routeParams.project + "/browse/deployments/" + deploymentConfig.metadata.name
+            });
+          }
+
+          var deployment = $scope.attach.deployment;
+          if (deployment) {
+            var deploymentVersion = $filter("annotation")(deployment, "deploymentVersion");
+            $scope.breadcrumbs.splice(2, 0, {
+              title: deploymentVersion ? "#" + deploymentVersion : deployment.metadata.name
+            });
+            var deploymentDeploymentConfigName = $filter("annotation")(deployment, "deploymentConfig");
+            if (deploymentDeploymentConfigName) {
+              $scope.breadcrumbs[2].link = "project/" + $routeParams.project + "/browse/deployments/" + deploymentDeploymentConfigName + "/" + deployment.metadata.name;
+              $scope.breadcrumbs.splice(2, 0, {
+                title: deploymentDeploymentConfigName,
+                link: "project/" + $routeParams.project + "/browse/deployments/" + deploymentDeploymentConfigName
+              });
+            }
+          }
+        };
+
+        var orderByDisplayName = $filter('orderByDisplayName');
+        var getErrorDetails = $filter('getErrorDetails');
```

Comment:
Move these above where they're first called in case the DataService callback ever happens immediately.